tgz.go

```diff
@@ -67,6 +103,10 @@ func (a *Archive) AddFileByContent(src []byte, dest string) error {
 }
 
 func (a *Archive) Close() {
+	close(a.WriteQueue)
+
+	<-a.Finished
```

Comment:
it doesn't send the true down this channel until the after the last call to addFileByContent has returned, so I don't see how it could get here before the last write is completed?